src/graph.cpp

```diff
@@ -118,6 +122,23 @@ void Graph::generate_delta_node(node_id_t node_n, uint64_t node_seed, node_id_t
 void Graph::batch_update(node_id_t src, const std::vector<node_id_t> &edges, Supernode *delta_loc) {
   if (update_locked) throw UpdateLockedException();
 
+  for (auto it = edges.begin(); dsu_valid && it != edges.end(); ++it) {
```

Comment:
Batch update is performed in parallel from many graph workers. Is this code thread safe?

Reply:
Actually, it would probably be better to do the DSU update in graph::update. I know this post buffering thing was my idea but the main computation we want to avoid is flushing the buffers during queries. We need this in Graph::update if that'll happen.